modules/lead/product-jenkins/artifactory.tf

```diff
@@ -1,34 +1,12 @@
-data "kubernetes_secret" "jenkins_artifactory_credential" {
-  count    = var.enable_artifactory ? 1 : 0
-  provider = kubernetes.toolchain
-  metadata {
-    name      = "jenkins-artifactory-credential"
-    namespace = "toolchain"
-  }
+resource "random_password" "robot_password" {
+  length           = 16
+  special          = true
+  override_special = "_%@"
 }
 
-data "template_file" "maven_settings" {
-  count    = var.enable_artifactory ? 1 : 0
-  template = file("${path.module}/artifactory-maven-settings.tpl")
-
-  vars = {
-    username = data.kubernetes_secret.jenkins_artifactory_credential[0].data.username
-    password = data.kubernetes_secret.jenkins_artifactory_credential[0].data.password
-  }
-}
-
-resource "kubernetes_secret" "jenkins_artifactory_maven_settings" {
-  count    = var.enable_artifactory ? 1 : 0
-  provider = kubernetes.toolchain
-  metadata {
-    name      = "jenkins-artifactory-maven-settings"
-    namespace = module.toolchain_namespace.name
-  }
-
-  data = {
-    "settings.xml" = data.template_file.maven_settings[0].rendered
-  }
-
-  type = "Opaque"
+resource "artifactory_user" "robot" {
+  count      = var.enable_artifactory ? 1 : 0
+  name       = "robot-${var.product_name}" 
+  email      = "robot-${var.product_name}@liatrio.com"
+  password   = data.random_password.robot_password.result
```

Comment:
is there a reason these changes are within the `artifactory.tf` file and not a new file?  is the plan to completely deprecate artifactory usage within lead-terraform?

Reply:
Maybe?

Reply:
if so, I'd still prefer changes like that to show up in another PR, and not do some of them in this one, then follow up with the rest.

can we add these changes in a new file?  also, how do these user credentials get seeded to jenkins?

Reply:
Artifactory already is deprecated or it was back when we did the big refactor. This was a leftover file that I repurposed.

Reply:
gotcha, can we rename the file then so it's clear that this is specifically for JCR?